### PR TITLE
Fix the example in documentation for `get_dremel_data()`

### LINKED ISSUE #11396: The example in the documentation for `get_dremel_data()` seems incorrect at line#1764
I know this isn't part of the review, but the example in the documentation for `get_dremel_data()` seems incorrect at line#1738 (now [#1764](https://github.com/rapidsai/cudf/pull/11328/files#diff-e3891cd717ca174e53dd11bbf812ad3f13034d4c1a4626ea751f05bbed8470d5R1764)):
```c++
 * Given a LIST column of type `List<List<int>>` like so:
 * ```
 * col = {
 *    [],
 *    [[], [1, 2, 3], [4, 5]],
 *    [[]]
 * }
 * ```
 * We can represent it in cudf format with two level of offsets like this:
 * ```
 * Level 0 offsets = {0, 0, 3, 5, 6}
 * Level 1 offsets = {0, 0, 3, 5, 5}
 * Values          = {1, 2, 3, 4, 5}
 * ```
```
The `Level 0` offset values can't exceed `4`, since `Level 1` has only 4 ranges (i.e. 5 offsets).
I'll try make sense of this at a later date, but I'm not sure I could follow along.

_Originally posted by @mythrocks in https://github.com/rapidsai/cudf/pull/11328#discussion_r932727013_

### PR DESCRIPTION
**Pull request by mhaseeb123**
## Description

Closes #11396. Fixes the example in the documentation of `get_dremel_data()`

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/include/cudf/lists/detail/dremel.hpp [hunk 1]
```diff
@@ -74,7 +74,7 @@ struct dremel_data {
  *
  * http://www.goldsborough.me/distributed-systems/2019/05/18/21-09-00-a_look_at_dremel/
  * https://akshays-blog.medium.com/wrapping-head-around-repetition-and-definition-levels-in-dremel-powering-bigquery-c1a33c9695da
- * https://blog.twitter.com/engineering/en_us/a/2013/dremel-made-simple-with-parquet
+ * https://blog.x.com/engineering/en_us/a/2013/dremel-made-simple-with-parquet
```
Comment (mhaseeb123) [inline cpp/include/cudf/lists/detail/dremel.hpp:77]: Previous link was stale

**Review by davidwendt — APPROVED** [review]

File: cpp/include/cudf/lists/detail/dremel.hpp [hunk 2]
```diff
@@ -102,16 +102,17 @@ struct dremel_data {
...
- * col = {[], [[], [1, 2, 3], [4, 5]], [[]]}
- * def = { 0    1,  2, 2, 2,   2, 2,     1 }
- * rep = { 0,   0,  0, 2, 2,   1, 2,     0 }
+ * col =            {[], [[], [1, 2, 3], [4, 5]], [[]]}
+ * dremel_offsets = { 0,  1,                       7, 8}
+ * def_levels     = { 0   1,   2, 2, 2,   2, 2,    1 }
+ * rep_levels     = { 0,  0,   1, 2, 2,   1, 2,    0 }
  * ```
  *
  * Since repetition and definition levels arrays contain a value for each empty list, the size of
```
Comment (mhaseeb123) [inline cpp/include/cudf/lists/detail/dremel.hpp:118]: Reviewers: Would love an extra set of eyes at the explanation of the algorithm at L120-136 if you could. Thanks!

File: cpp/include/cudf/lists/detail/dremel.hpp [hunk 3]
```diff
@@ -102,16 +102,17 @@ struct dremel_data {
  * ```
  * We can represent it in cudf format with two level of offsets like this:
  * ```
- * Level 0 offsets = {0, 0, 3, 5, 6}
+ * Level 0 offsets = {0, 0, 3, 4}
```
Comment (vuule) [inline cpp/include/cudf/lists/detail/dremel.hpp:105]: this looks right :+1:

File: cpp/include/cudf/lists/detail/dremel.hpp [hunk 4]
```diff
@@ -102,16 +102,17 @@ struct dremel_data {
...
- * correspond to the data values:
+ * The desired result of this function is the dremel offsets, and repetition and definition level
+ * values that correspond to the data values:
  * ```
- * col = {[], [[], [1, 2, 3], [4, 5]], [[]]}
- * def = { 0    1,  2, 2, 2,   2, 2,     1 }
- * rep = { 0,   0,  0, 2, 2,   1, 2,     0 }
+ * col =            {[], [[], [1, 2, 3], [4, 5]], [[]]}
+ * dremel_offsets = { 0,  1,                       7, 8}
+ * def_levels     = { 0   1,   2, 2, 2,   2, 2,    1 }
```
Comment (vuule) [inline cpp/include/cudf/lists/detail/dremel.hpp:114]:
```suggestion
 * def_levels     = { 0,  1,   2, 2, 2,   2, 2,    1 }
```

**Review by vuule — CHANGES_REQUESTED** [review]

File: cpp/include/cudf/lists/detail/dremel.hpp (hunk 2 above)
Comment (vuule) [reply cpp/include/cudf/lists/detail/dremel.hpp:118]: Looks good, I'm just confused as to why we included the algorithm description in the public header :)
Not a concern for this PR

**Review by vuule — APPROVED** [review]

File: cpp/include/cudf/lists/detail/dremel.hpp [hunk 5]
```diff
@@ -58,7 +58,7 @@ struct dremel_data {
 };
 
 /**
- * @brief Get the dremel offsets and repetition and definition levels for a LIST column
+ * @brief Get the dremel offsets, and repetition and definition levels for a LIST column
```
Comment (hyperbolic2346) [inline cpp/include/cudf/lists/detail/dremel.hpp:61]:
```suggestion
 * @brief Get the dremel offsets, repetition, and definition levels for a LIST column
```

I feel like we're Oxford comma people, right?

File: cpp/include/cudf/lists/detail/dremel.hpp [hunk 6]
```diff
@@ -102,16 +102,17 @@ struct dremel_data {
...
  * We can represent it in cudf format with two level of offsets like this:
  * ```
- * Level 0 offsets = {0, 0, 3, 5, 6}
+ * Level 0 offsets = {0, 0, 3, 4}
  * Level 1 offsets = {0, 0, 3, 5, 5}
  * Values          = {1, 2, 3, 4, 5}
  * ```
- * The desired result of this function is the repetition and definition level values that
- * correspond to the data values:
+ * This function returns the dremel offsets, and repetition and definition level
```
Comment (hyperbolic2346) [inline cpp/include/cudf/lists/detail/dremel.hpp:109]:
```suggestion
 * This function returns the dremel offsets, repetition, and definition level
```

**Review by hyperbolic2346 — APPROVED** [review]
Some small nits.

File: cpp/include/cudf/lists/detail/dremel.hpp (hunk 5 above)
Comment (mhaseeb123) [reply cpp/include/cudf/lists/detail/dremel.hpp]: Yes we are!

Comment (vuule) [reply cpp/include/cudf/lists/detail/dremel.hpp]: Doesn't it change the meaning? AFAICT, this now needs to be "dremel offsets, repetition levels, and definition levels"

Comment (mhaseeb123) [reply cpp/include/cudf/lists/detail/dremel.hpp]: Yes it does and that's exactly what I updated it to!

**Review by mythrocks — APPROVED** [review]

File: cpp/include/cudf/lists/detail/dremel.hpp (hunk 3 above)
Comment (GregoryKimball) [reply cpp/include/cudf/lists/detail/dremel.hpp:105]: 😱

**Pull request merged**